crates/oxc_parser/src/lexer/simd.rs

```diff
@@ -84,3 +83,118 @@ impl SkipWhitespace {
         self.offset += advance_by as usize;
     }
 }
+
+pub struct SkipMultilineComment<'a> {
+    /// Total offset
+    pub offset: usize,
+
+    /// Found multiline comment end '*/'?
+    pub found: bool,
+
+    /// Found newline inside the comment?
+    pub newline: bool,
+
+    /// Remaining char bytes from the lexer
+    remaining: &'a [u8],
+
+    star: SimdVec,
+    slash: SimdVec,
+    lf: SimdVec,
+    cr: SimdVec,
+    lsps: SimdVec,
+}
+
+impl<'a> SkipMultilineComment<'a> {
+    pub fn new(newline: bool, remaining: &'a [u8]) -> Self {
+        Self {
+            offset: 0,
+            found: false,
+            newline,
+            remaining,
+            star: SimdVec::splat(b'*'),
+            slash: SimdVec::splat(b'/'),
+            lf: SimdVec::splat(b'\n'),
+            cr: SimdVec::splat(b'\r'),
+            lsps: SimdVec::splat(226),
+        }
+    }
+
+    pub fn simd(mut self, remaining: &[u8]) -> Self {
+        let (chunks, remainder) = remaining.as_chunks::<ELEMENTS>();
+
+        for chunk in chunks {
+            self.check(chunk, chunk.len());
+            if self.found {
+                return self;
+            }
+        }
+
+        if !remainder.is_empty() {
+            // Align the last chunk for avoiding the use of a scalar version
+            let mut chunk = [0; ELEMENTS];
+            let len = remainder.len();
+            chunk[..len].copy_from_slice(remainder);
+            self.check(&chunk, len);
+        }
+
+        self
+    }
+
+    /// Check and compute state for a single chunk
+    /// `chunk_len` can be < ELEMENTS for the last chunk
+    fn check(&mut self, chunk: &[u8], chunk_len: usize) {
+        let s = SimdVec::from_slice(chunk);
+
+        let any_star = s.simd_eq(self.star);
+        let any_slash = s.simd_eq(self.slash);
+        let star_mask = any_star.to_bitmask();
+        let slash_mask = any_slash.to_bitmask();
```

Comment:
I have found that to_bitmask on ARM NEON can be cause parsing to be *slower* compared to a scalar version. My block comment parser only uses SIMD on x86, not ARM. Do you run your benchmarks on ARM NEON or just Intel SSE?

On x86_64, to_bitmask is one instruction. On AArch64, there is no instruction and to_bitmask needs to be emulated. See: https://godbolt.org/z/531oEo5d1 It looks like the Rust compiler currently does a terrible job on AArch64. You can definitely do better by hand, but it still might not be faster than the scalar version.